### DISCUSSION TITLE
why not work create by multi where?

### DISCUSSION THREAD

**Discussion by ashkankiani** [prisma/prisma#24723] [Q&A] [answered]
### Question

code is not insert
```js            
const response = await prisma.draft.create({
  where: {
    serviceId: {
      not: body.serviceId,
    },
    providerId: {
      not: body.providerId,
    },
  },
  data: body,
});
```
Not with the UPSERT method either
```js
const response = await prisma.draft.upsert({
  where: {
    serviceId: {
      not: body.serviceId,
    },
    providerId: {
      not: body.providerId,
    },
    date: {
      not: body.date,
    },
    time: {
      not: body.time,
    },
  },
  update: {},
  create: body,
})
````

```sql
-- CreateTable
CREATE TABLE `Draft` (
    `id` INTEGER NOT NULL AUTO_INCREMENT,
    `createEpoch` BIGINT NOT NULL,
    `serviceId` INTEGER NOT NULL,
    `providerId` INTEGER NOT NULL,
    `userId` INTEGER NULL,
    `dateTimeStartEpoch` BIGINT NOT NULL,
    `dateTimeEndEpoch` BIGINT NOT NULL,
    `date` VARCHAR(191) NOT NULL,
    `time` VARCHAR(191) NOT NULL,
    `createdAt` DATETIME(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
    `updatedAt` DATETIME(3) NOT NULL,
    PRIMARY KEY (`id`)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @ashkankiani 👋 

Thank you for raising this question.

Looking at the first code where you are using the Prisma Client `create` method, I can see you are using a `where` clause. Please note that the **create method does not support a where clause**. It is used solely for creating new records. If you need to ensure that a record does not already exist before creating it, you should use the `upsert` method instead. See this [section](https://www.prisma.io/docs/orm/reference/prisma-client-reference#create) of the documentation for the options available in the  `create` method.

For the second code using the [`upsert`](https://www.prisma.io/docs/orm/reference/prisma-client-reference#upsert) method,  you should note that the `upsert` method requires a unique identifier in the `where` clause to determine whether to update an existing record or create a new one. The where clause should reference a unique field or a combination of fields that uniquely identify a record. Prisma's `upsert` method requires at least one unique field to be specified in the [`where`](https://www.prisma.io/docs/orm/reference/prisma-client-reference#userwhereuniqueinput-considerations) clause, but it does not have to be a single unique identifier. You can use boolean operators like AND, OR, and NOT in conjunction with unique fields, as long as at least one unique field is specified outside of these boolean operators.
```js
const response = await prisma.draft.upsert({
  where: {
    id: body.id, // Assuming `id` is a unique identifier
    AND: [
      { serviceId: { not: body.serviceId } },
      { providerId: { not: body.providerId } },
      { date: { not: body.date } },
      { time: { not: body.time } },
    ],
  },
  update: {
    // Fields to update if the record exists
    serviceId: body.serviceId,
    providerId: body.providerId,
    date: body.date,
    time: body.time,
    // Add other fields as necessary
  },
  create: {
    // Fields to create if the record does not exist
    serviceId: body.serviceId,
    providerId: body.providerId,
    userId: body.userId,
    dateTimeStartEpoch: body.dateTimeStartEpoch,
    dateTimeEndEpoch: body.dateTimeEndEpoch,
    date: body.date,
    time: body.time,
    createdAt: new Date(),
    updatedAt: new Date(),
  },
});
```

**Comment by ashkankiani** [comment, marked as answer]
I am thankful with your guidance. Exactly my problem was that I didn't have a unique ID according to my condition. I needed to add to the database as long as my 4 columns are unique. On the other hand, I did not have the conditions to have a unique one.
I will find a solution to the problem. It was enough to put my 4 columns as the Primary key and not have a Primary key like id anymore. In this way, when data is added to the database, I make sure that it is not unique or duplicated.

`
model Draft {
  // id Int @id @default(autoincrement())

  createEpoch BigInt

  service   Services @relation(fields: [serviceId], references: [id])
  serviceId Int

  provider   Providers @relation(fields: [providerId], references: [id])
  providerId Int

  user   Users? @relation(fields: [userId], references: [id])
  userId Int?

  dateTimeStartEpoch BigInt
  dateTimeEndEpoch   BigInt
  date               String
  time               String

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@id([serviceId, providerId, date, time])
}
`
